src/Handler/NewScream.hs

```diff
@@ -27,13 +29,22 @@ data ScreamFields = ScreamFields
     , imageField :: Maybe FileInfo
     }
 
-fromScreamFields :: ScreamFields -> Handler Scream
-fromScreamFields f = do
+parseImages :: ScreamFields -> ScreamId -> Handler [Image]
+parseImages f sid = do
+    images <- S3.uploadImages $ maybeToList $ imageField f
+    return $ fmap createImage images
+  where
+      createImage (fileInfo, url) = Image
+          { imageScreamId = sid
```

Comment:
you could leave out the record syntax if you prefer here `Image sid (fileName fileInfo) url`. Not sure you're gaining much from explicitness here.